utilities/ride-model-util/src/main/java/com/adobe/ride/utilities/model/ModelObject.java

```diff
@@ -436,16 +441,15 @@ public static ModelPropertyType getModelPropertyType(JSONObject object)
         returnValue = ModelPropertyType.URI;
       } else if (format.equals("uri-reference")) {
         returnValue = ModelPropertyType.URI_REF;
-      } else if (object.containsKey("pattern")) {
-        returnValue = ModelPropertyType.PATTERN;
-      } else if (type.equals("string")) {
-        returnValue = ModelPropertyType.eval(format);
-      }
+      } 
     } else if (object.containsKey("pattern")) {
       returnValue = ModelPropertyType.PATTERN;
-    } else {
-      returnValue = ModelPropertyType.eval(object.get("type").toString());
+    } 
+    	
+    if (returnValue == null && !"noType".equals(type)) {
+    	returnValue = ModelPropertyType.eval(type);
```

Comment:
This change I believe looks good.  Although I have to mention that it doesn't completely solve the problem defined in the issue that spurred this PR.  Some additional values will have to be added here to solve that OAS issue:
https://github.com/adobe/ride/blob/develop/utilities/ride-model-util/src/main/java/com/adobe/ride/utilities/model/types/ModelPropertyType.java

And code will have to be added to generateNodeValue to handle those additional formats.  However, these can be added as they are needed.

refenction: {"format":"int64"...}

Reply:
You are right. However, it is left for the tools to decide whether to support all formats or to fall back to specified type - https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#dataTypeFormat

The problem with the earlier approach is that we have to keep maintaining the `ModelPropertyType` and the methods like `generateNodeValue`, which I believe is a bit maintenance headache.

Reply:
Well, I suppose subclassing and overriding could get you there, but contributing to the source (as you have done, and thank you for doing so :) ), will help make this better for all adopters/contributors/implementors.  I guess it's probably a balance.  People should add ones that they feel will help the overall project, and subclass for things that don't rise to that level.